pkgs/build-support/trivial-builders.nix

```diff
@@ -559,10 +559,22 @@ rec {
    *                 passthru.tests.greeting = callPackage ./test { };
    *               } ./myscript.sh;
    */
-  makeSetupHook = { name ? "hook", deps ? [], substitutions ? {}, meta ? {}, passthru ? {} }: script:
+  makeSetupHook =
+    { name ? "hook"
+    , substitutions ? { }
+    , meta ? { }
+    , passthru ? { }
+    , deps ? [ ]
+    # hooks go in nativeBuildInput so these will be nativeBuildInput
+    , propagatedBuildInputs ? [ ]
+    # these will be buildInputs
+    , depsTargetTargetPropagated ? [ ]
+    }: script:
     runCommand name
       (substitutions // {
         inherit meta;
+        inherit depsTargetTargetPropagated;
+        propagatedBuildInputs = assert (lib.assertMsg (lib.isList deps) "deps argument to makeSetupHook must be a list. content: ${toString deps}"); deps ++ propagatedBuildInputs;
```

Comment:
I would go for a soft deprecation (using a `trace`) first so that external consumers can adapt.

And while at it we might as well deprecate `deps` in favour of `propagatedBuildInputs`.

Reply:
> I would go for a soft deprecation (using a trace) first so that external consumers can adapt.

done

Reply:
and deps deprecated